cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -271,62 +271,12 @@ angular.module(PKG.name + '.commons')
       }
 
       angular.forEach($scope.nodes, function (node) {
-        if (node.type !== 'condition') {
-          let sourceObj = {
-            isSource: true,
-            filter: function(event) {
-              // we need this variable because when the user clicks on the endpoint circle, multiple
-              // 'mousedown' events are fired
-              if (event.target.className.indexOf('endpoint-circle') !== -1 && !endpointClicked) {
-                endpointClicked = true;
-              }
-
-              return event.target.className.indexOf('endpoint-circle') !== -1;
-            },
-            connectionType: 'basic'
-          };
-          if (vm.isDisabled) {
-            sourceObj.enabled = false;
-          }
-
-          vm.instance.makeSource(node.name, sourceObj);
-
+        if (node.type !== 'condition' && node.type !== 'splittertransform') {
+          initNormalNode(node.name);
+        } else if (node.type === 'condition') {
+          initConditionNode(node.name);
         } else {
-          if (conditionNodes.indexOf(node.name) !== -1) {
-            return;
-          }
-          let newTrueEndpoint = vm.instance.addEndpoint(node.name, {
-            anchor: 'Right',
-            cssClass: 'condition-endpoint condition-endpoint-true',
-            isSource: true,
-            maxConnections: -1,
-            endpoint: 'Dot',
-            uuid: 'True' + node.name,
-            connectorStyle: vm.conditionTrueConnectionStyle,
-            overlays: [
-              [ 'Label', { label: 'Yes', id: 'yesLabel', location: [0.5, -0.55], cssClass: 'condition-label' } ]
-            ]
-          });
-          newTrueEndpoint.hideOverlay('yesLabel');
-
-          let newFalseEndpoint = vm.instance.addEndpoint(node.name, {
-            anchor: 'Bottom',
-            cssClass: 'condition-endpoint condition-endpoint-false',
-            isSource: true,
-            maxConnections: -1,
-            endpoint: 'Dot',
-            uuid: 'False' + node.name,
-            connectorStyle: vm.conditionFalseConnectionStyle,
-            overlays: [
-              [ 'Label', { label: 'No', id: 'noLabel', location: [0.5, -0.55], cssClass: 'condition-label' } ]
-            ]
-          });
-          newFalseEndpoint.hideOverlay('noLabel');
-
-          addListenersForEndpoint(newTrueEndpoint, 'right', node.name, 'yesLabel');
-          addListenersForEndpoint(newFalseEndpoint, 'bottom', node.name, 'noLabel');
-
-          conditionNodes.push(node.name);
+          initSplitterNode(node);
```

Comment:
might want to reverse this if-else block so that the `else` clause is the normal node. That way, we don't have to keep appending the `if` condition for the default nodes.

Reply:
yep, good idea